test/e2e/framework/helpers/rest_api_utils.go

```diff
@@ -0,0 +1,114 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+package helpers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/apache/incubator-yunikorn-k8shim/test/e2e/framework/cfg_manager"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+type RClient struct {
+	BaseURL   *url.URL
+	UserAgent string
+
+	httpClient *http.Client
+}
+
+func (c *RClient) newRequest(method, path string, body interface{}) (*http.Request, error) {
+	rel := &url.URL{Path: path}
+	if c.BaseURL == nil {
+		c.BaseURL = &url.URL{Host: GetYKHost(),
+			Scheme: GetYKScheme()}
+	}
+	u := c.BaseURL.ResolveReference(rel)
+	if c.httpClient == nil {
+		c.httpClient = http.DefaultClient
+	}
+	if len(c.UserAgent) == 0 {
+		c.UserAgent = "Golang_Spider_Bot/3.0"
```

Comment:
Not sure what does this does. Do we need this?

Reply:
This is just to set the user agent string, best practice and better debugging!